str/str_iterative_hipstr_runner.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python3
+# pylint: disable=import-error
+
+"""
+This script uses HipSTR to call STRs on WGS cram files, using the joint calling option. 
+For example:
+analysis-runner --access-level test --dataset hgdp --description 'hipstr run' --output-dir 'str/sensitivity-analysis/hipstr' str_iterative_hipstr_runner.py --variant-catalog=gs://cpg-hgdp-test/str/untrimmed_coordinates_resources/hg38_hipstr_catalog_untrimmed_coordinates.bed --output-file-name=hipster_90_genomes --dataset=hgdp HGDP00511
+
+Required packages: sample-metadata, hail, click, os
+pip install sample-metadata hail click
+
+"""
+import logging
+
+
+import click
+from sample_metadata.apis import AnalysisApi, SampleApi
+from sample_metadata.model.analysis_type import AnalysisType
+from sample_metadata.model.analysis_query_model import AnalysisQueryModel
+from sample_metadata.models import AnalysisStatus
+from cpg_utils.config import get_config
+from cpg_utils.hail_batch import output_path
+from cpg_workflows.batch import get_batch
+
+config = get_config()
+
+SAMTOOLS_IMAGE = config['images']['samtools']
+HIPSTR_IMAGE = config['images']['hipstr']
+BCFTOOLS_IMAGE = config['images']['bcftools']
+
+
+# inputs:
+@click.option('--variant-catalog', help='Full path to HipSTR Variants catalog')
+@click.option('--dataset', help='dataset eg tob-wgs')
+@click.argument('external-wgs-ids', nargs=-1)
+@click.option('--output-file-name', help='Output file name without file extension')
+@click.command()
+def main(
+    variant_catalog, dataset, external_wgs_ids, output_file_name
+):  # pylint: disable=missing-function-docstring
+    b = get_batch()
+    #Create HipSTR job 
+    hipstr_job = b.new_job(name=f'HipSTR running')
+    hipstr_job.image(HIPSTR_IMAGE)
+    hipstr_job.storage('375G')
+    hipstr_job.cpu(16)
```

Comment:
This was the memory profile for 91 samples: https://batch.hail.populationgenomics.org.au/batches/421353/jobs/1
I'm planning to run it on 150..

Reply:
How do I calculate how much memory and CPUs I need?

Reply:
That batch looks like it's only using 1 CPU (single threaded), but it's using a bit of memory. Hard to really gauge how much memory it will use, it might be opportunistically allocating memory (eg: it will happily use less memory).

You might consider using the `highmem` machine, then either 4 / 8 cores (which gives 26GB / 52GB RAM).

Reply:
How do I set the job to `highmem` exactly? I tried looking for examples in the popgen repo and Hail Batch documentation but couldn't find an example for this context.

Reply:
This should do it :)
```suggestion
    hipstr_job.cpu(4)
    j.memory('highmem')
```